### PR TITLE
Fix Content Collections not loading config file when there are spaces in the folder tree

### LINKED ISSUE #5961: Content Collections can't load `config.ts` file when there are spaces in the folder tree
### What version of `astro` are you using?

2.0.0

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

pnpm

### What operating system are you using?

Ubuntu 22.04

### Describe the Bug

When there are spaces in the folder tree leading to the config file loaded by the content collections API, loading the config file fails

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-moeeyn?file=package.json&on=stackblitz

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by MoustaphaDev**
## Changes

Fix #5961 

## Testing

TODO

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
Can you add a test? Also I think it should be done like its done here: https://github.com/withastro/astro/blob/main/packages/astro/src/core/render/dev/index.ts#L60

using `fileURLToPath()`

**Comment by MoustaphaDev** [comment]
> Also I think it should be done like its done here: https://github.com/withastro/astro/blob/main/packages/astro/src/core/render/dev/index.ts#L60
> 
> using `fileURLToPath()`

Ah ok, testing that

**Comment by MoustaphaDev** [comment]
@matthewp, some debugging and I think `decodeURIComponent` would do it here, in the console it looks like the logged file is indeed a path not a file url

File: packages/astro/src/content/utils.ts [hunk 1]
```diff
@@ -228,7 +228,7 @@ export async function loadContentConfig({
 	}
 	try {
 		unparsedConfig = await tempConfigServer.ssrLoadModule(
-			decodeURIComponent(contentPaths.config.pathname)
+			fileURLToPath(contentPaths.config.pathname)
```
Comment (MoustaphaDev) [inline packages/astro/src/content/utils.ts:231]: Meant to do 
```js
new URL(fileURLToPath(contentPaths.config)).pathname
```
But that results in the majority of tests to fail. I'll revert back to using `decodeURIComponent`

Comment (matthewp) [reply packages/astro/src/content/utils.ts]: Are tests failing with just `fileURLToPath(contentPaths.config.pathname)`? (without the URL wrapper)?

Comment (MoustaphaDev) [reply packages/astro/src/content/utils.ts]: Yes, in both cases

Comment (matthewp) [reply packages/astro/src/content/utils.ts]: Oh sorry, I would expect it to be `fileURLToPath(contentPaths.config)`. config is a file URL. That converts it to a regular path which Vite can take.

Comment (matthewp) [reply packages/astro/src/content/utils.ts]: Essentially you almost never need/want to use `.pathname` anywhere that takes a file path.

Comment (MoustaphaDev) [reply packages/astro/src/content/utils.ts]: Ah thanks for your help, that did it!

**Review by matthewp — APPROVED** [review]
looks great, thanks!

**Review by bholmesdev — APPROVED** [review]
Nice fix, _and_ with tests! Love the `config`dence 👏

**Comment by matthewp** [comment]
The failing test is the flaky one, will restart.

**Comment by MoustaphaDev** [comment]
> The failing test is the flaky one, will restart.

Oops I thought the same, merged anyway as it's been ignore a lot of time
I can revert

**Pull request merged**